app/controllers/conversions/projects_controller.rb

```diff
@@ -22,6 +22,37 @@ def create
     end
   end
 
+  def edit
+    @project = Project.find(params[:id])
+    authorize @project
+
+    @project = Conversion::EditProjectSharepointLinkForm.new(project: project)
+    @project_id = params[:id]
+  end
+
+  def update
+    authorize project
+    @project = Conversion::EditProjectSharepointLinkForm.new(project: project, args: sharepoint_link_params)
+    @project_id = params[:id]
+
+    if @project.save
+      redirect_to project_information_path(project), notice: I18n.t("project.update.success")
+    else
+      render :edit
+    end
+  end
+
+  private def project
+    Project.find(params[:id])
```

Comment:
If you change this to `@project ||= Project.find(params[:id])` it will memoize it and you can get rid of any other calls to `Project.find(params[:id])` elsewhere